pkg/types/kubevirt/validation/machinepool.go

```diff
@@ -0,0 +1,40 @@
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types/kubevirt"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *kubevirt.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.CPU != 0 {
+		if p.CPU < 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpu"), p.CPU, "CPU must be positive"))
+		}
+	}
+
+	if p.StorageSize != "" {
+		storageQuantity, err := resource.ParseQuantity(p.StorageSize)
+
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("storage"), p.StorageSize, "Storage size must be of Quantity type format"))
+		} else if storageQuantity.Sign() != 1 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("storage"), p.StorageSize, "Storage size must be positive value"))
+		}
+	}
+
+	if p.Memory != "" {
+		memoryQuantity, err := resource.ParseQuantity(p.Memory)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("memory"), p.Memory, "Memory must be of Quantity type format"))
```

Comment:
err isn't passed to the append so you lose the cause

Reply:
Fixed